ide-plugin/src/org/jetbrains/plugins/ruby/ruby/codeInsight/types/RubyStatTypeProviderImpl.java

```diff
@@ -307,7 +384,7 @@ private static MethodInfo findMethodInfo(@NotNull final PsiElement callElement)
             arg = ((RAssoc) arg).getValue();
         }
 
-        final Set<String> argTypeNames = new HashSet<>();
+        final Set<String> argTypeNames = new LinkedHashSet<>();
```

Comment:
Why is order imporant here?

Reply:
You can sort stuff in debug messages; now you add memory and (little) cpu overhead for common code